src/movies.js

```diff
@@ -1,26 +1,78 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+//console.log('movies: ', movies);
+
+const movies = require("./data");
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  let directors = array.map((movies) => movies.director)
+  return directors
+}
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
 
+
+// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(arr) {
+  let stevenMovies = arr.filter((movie) => movie.genre.includes('Drama') && movie.director === "Steven Spielberg");
+  return stevenMovies.length
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  if (arr.length === 0) {
+    return 0
+  }
+  let average = arr.reduce(function(acc, value) {
+    if(!value.score){
+      value.score = 0 
+    }
+  return acc + value.score;
+  }, 0);
+  return Number((average / arr.length).toFixed(2))
+}
+
+console.log(scoresAverage(movies))
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(arr) {
+  let dramaMovies = arr.filter((movie) => movie.genre.includes('Drama'));
+  return scoresAverage(dramaMovies);
+}
+
+
+
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(arr) {
+  let newArr = [];
+  const ordered = arr.sort(function(a, b) {
+  if (a.year < b.year) return -1;
+  if (a.year > b.year) return 1;
+  if (a.year === b.year) return 0;
+   // a.title.localeCompare(b.title) ??
+  })
+  return ordered
```

Comment:
You should return a new array, you are returning the array passed as a parameter of your function